Testscripts/Linux/SetupHbKernel.sh

```diff
@@ -141,18 +152,114 @@ function Main() {
 			LogMsg "$?: Added resume=$sw_uuid in /etc/default/grub file"
 		fi
 
+		_entry=$(cat /etc/default/grub | grep 'GRUB_HIDDEN_TIMEOUT=')
+		if [ -n "$_entry" ]; then
+			sed -i -e "s/GRUB_HIDDEN_TIMEOUT=*.*/GRUB_HIDDEN_TIMEOUT=30/g" /etc/default/grub
+			LogMsg "$?: Updated GRUB_HIDDEN_TIMEOUT value with 30"
+		else
+			echo 'GRUB_HIDDEN_TIMEOUT=30' >> /etc/default/grub
+			LogMsg "$?: Added GRUB_HIDDEN_TIMEOUT=30 in /etc/default/grub file"
+		fi
+
+		_entry=$(cat /etc/default/grub | grep 'GRUB_TIMEOUT=')
+		if [ -n "$_entry" ]; then
+			sed -i -e "s/GRUB_TIMEOUT=.*/GRUB_TIMEOUT=30/g" /etc/default/grub
+			LogMsg "$?: Updated GRUB_TIMEOUT value with 30"
+		else
+			echo 'GRUB_TIMEOUT=30' >> /etc/default/grub
+			LogMsg "$?: Added GRUB_TIMEOUT=30 in /etc/default/grub file"
+		fi
+
+		# VM Gen is 2
 		if [[ "$os_GENERATION" == "2" ]];then
 			grub_cfg="/boot/efi/EFI/redhat/grub.cfg"
 		else
-			grub_cfg="/boot/grub2/grub.cfg"
+			# VM Gen is 1
+			if [ -f /boot/grub2/grub.cfg ]; then
+				grub_cfg="/boot/grub2/grub.cfg"
+			else
+				grub_cfg="/boot/grub/grub.cfg"
+			fi
+		fi
+		# grub2-mkconfig shows the image build problem in RHEL/CentOS, so we use alternative.
+		#grub2-mkconfig -o ${grub_cfg}
+		#LogMsg "$?: Run grub2-mkconfig -o ${grub_cfg}"
+		ls /boot/vmlinuz* > new_state.txt
+		vmlinux_file=$(diff old_state.txt new_state.txt | tail -n 1 | cut -d ' ' -f2)
+		if [ -f $vmlinux_file ]; then
+			original_args=$(grubby --info=0 | grep -i args | cut -d '"' -f 2)
+			LogMsg "Original boot parameters $original_args"
+			grubby --args="$original_args resume=$sw_uuid" --update-kernel=$vmlinux_file
+			
+			grubby --set-default=$vmlinux_file
+			LogMsg "Set $vmlinux_file to the default kernel"
+			
+			new_args=$(grubby --info=ALL)
+			LogMsg "Updated grubby output $new_args"
+			
+			grubby_output=$(grubby --default-kernel)
+			LogMsg "grubby default-kernel output $grubby_output"
+		else
+			LogErr "Can not set new vmlinuz file in grubby command. Expected new vmlinuz file, but found $vmlinux_file"
+			SetTestStateAborted
+			exit 0
+		fi
+	elif [[ "$DISTRO" =~ "sles" || "$DISTRO" =~ "suse" ]];then
+		# TODO: This part need another revision once we can access to SUSE repo.
```

Comment:
Indeed. It's quite fragile to rely on `rootdelay=` being in the kernel cmdline

Reply:
It is fragile, but this is one way we update with resume=UUID=xxxx for SWAP space. If there is another way, please let me know.